frontend/packages/console-dynamic-plugin-sdk/src/extensions/storage-provider.ts

```diff
@@ -0,0 +1,14 @@
+import { RouteComponentProps } from 'react-router';
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { ExtensionDeclaration, CodeRef } from '../types';
+
+export type StorageProvider = ExtensionDeclaration<
+  'console.storage-provider',
+  {
+    name: string;
+    Component: CodeRef<React.FC<RouteComponentProps>>;
```

Comment:
```suggestion
    Component: CodeRef<React.ComponentType<RouteComponentProps>>;
```

I wonder if it should be `Partial<RouteComponentProps>` instead.
cc @vojtechszocs

Reply:
Maybe we should re-export `RouteComponentProps` so that to use the extension, the dev doesn't need to also depend on `react-router`. This might be a better solution

Reply:
`export interface RouteComponentProps<Params extends { [K in keyof Params]?: string } = {}, C extends StaticContext = StaticContext, S = H.LocationState> {`

Sorry... re-exporting isn't going to solve the problem I thought about in terms of eliminating the dep on react-router. Not when the type itself has so many other react-router types in it.

Reply:
If you just go with what you originally had, we can also drop down to `Component: CodeRef<React.ComponentType<RouteComponentProps>>;` in the future.
So i think my first suggestion is the best.